lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/child/ChildController.java

```diff
@@ -61,19 +67,22 @@ public void onViewDisappear() {
     }
 
     public void onViewBroughtToFront() {
-        presenter.onViewBroughtToFront(resolveCurrentOptions());
+        presenter.onViewBroughtToFront(this, resolveCurrentOptions());
     }
 
     @Override
     public void applyOptions(Options options) {
         super.applyOptions(options);
-        presenter.applyOptions(this, resolveCurrentOptions());
+        final ParentController<?> parentController = getParentController();
+        final Options resolveCurrentOptions = parentController == null ? resolveCurrentOptions() :
+                parentController.resolveChildOptions(this);
+        presenter.applyOptions(this, resolveCurrentOptions);
```

Comment:
maybe we should have this fix in another PR?

Reply:
agree!